Doc set for VMFR (#110937)
openshift/openshift-docs: backup_and_restore/application_backup_and_restore/oadp-vmfr/oadp-vmfr-end-to-end.adoc

:_mod-docs-content-type: PROCEDURE
include::_attributes/common-attributes.adoc[]

[id="oadp-vmfr-end-to-end"]
= Recovering files from a virtual machine backup
:context: oadp-vmfr-end-to-end

[role="_abstract"]

> Recover files from a virtual machine (VM) backup by creating a backup, discovering which backups contain the target VM, and restoring individual files through SSH. This procedure walks through an end-to-end {oadp-short} virtual machine file restore (VMFR) workflow.

Reviewer: ```suggestion
Recover files from a virtual machine (VM) backup by creating a backup, discovering which backups contain the target VM, and restoring individual files through SSH. Complete this end-to-end {oadp-short} VM file restore (VMFR) workflow to quickly regain access to your data.
```

.Prerequisites

* You are logged in to the cluster with the `cluster-admin` role.
* You have installed the {oadp-short} Operator.
* You have {VirtProductName} installed and running on the cluster.

> * You have installed the `virtctl` CLI tool to access the VM.

Reviewer: ```suggestion
.Prerequisites

* You are logged in to the cluster with the `cluster-admin` role.
* You have installed the {oadp-short} Operator.
* {VirtProductName} is installed and running on the cluster.
* You have installed the `virtctl` CLI tool to access the VM.
```

----
+
where:

`<secret_name>`:: Specifies the name of the cloud credentials secret.
`<credentials_file_path>`:: Specifies the path to the file that contains the cloud storage credentials.

. Create a `DataProtectionApplication` (DPA) CR with the VMFR feature enabled and the `kubevirt` plugin:
+
[source,yaml]
----
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
metadata:
  name: vmfr-dpa
  namespace: openshift-adp
spec:
  backupLocations:
    - velero:
        credential:
          key: cloud
          name: <secret_name>
        default: true
        objectStorage:
          bucket: <bucket_name>
          prefix: velero
        provider: <provider>
  configuration:
    velero:
      defaultPlugins:
        - csi
        - openshift
        - kubevirt
        - <provider>
      disableFsBackup: false
    nodeAgent:
      enable: true
      uploaderType: kopia
  vmFileRestore:
    enable: true
----
+
where:

`<secret_name>`:: Specifies the name of the cloud credentials secret you created.
`<bucket_name>`:: Specifies the name of the object storage bucket.
`<provider>`:: Specifies the cloud provider plugin, such as `aws`, `gcp`, or `azure`.

> `vmFileRestore`:: Enable the VMFR feature by setting the `enable` field to `true`.

Reviewer: ```suggestion
`<secret_name>`:: Specifies the name of the cloud credentials secret you created.
`<bucket_name>`:: Specifies the name of the object storage bucket.
`<provider>`:: Specifies the cloud provider plugin, such as `aws`, `gcp`, or `azure`.
`vmFileRestore`:: Enables the VMFR feature by setting the `enable` field to `true`.
```

trying to keep all of these grammatically parallel
